Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -173,6 +173,20 @@ static void GenerateDSIException(u32 effective_address, bool write);
 template <XCheckTLBFlag flag, typename T, bool never_translate = false>
 static T ReadFromHardware(u32 em_address)
 {
+  const u32 em_address_start_page = em_address & ~(HW_PAGE_SIZE - 1);
+  const u32 em_address_end_page = (em_address + sizeof(T) - 1) & ~(HW_PAGE_SIZE - 1);
```

Comment:
Would it be worth creating a constant like PAGE_ALIGNMENT_MASK and using it here and in WriteToHardware?

Reply:
Done.